.travis.yml

```diff
@@ -22,13 +12,28 @@ matrix:
   allow_failures:
     - python: "3.7-dev"
 
+install:
+  - make travis-sysdeps
+  - ssh-keygen -t rsa -b 2048 -f ~/.ssh/id_rsa -P ""
+
+env:
+  global:
+    - PATH=/snap/bin:$PATH
+
+before_script:
+  - sudo addgroup lxd || true
+  - sudo usermod -a -G lxd $USER || true
+  - sudo ln -s /snap/bin/lxc /usr/bin/lxc
+
 # We set HOME because lxc needs somewhere writable to not crash during testing.
+# The "sudo -E" line is needed because the user was added to the lxd group which needs to be applied.
 script:
   - cp -R /home/travis/.ssh $TRAVIS_BUILD_DIR
-  - HOME=$TRAVIS_BUILD_DIR make travis
+  - sudo -E sudo -u $USER -E bash -c 'source ~/virtualenv/python$TRAVIS_PYTHON_VERSION/bin/activate; HOME=$TRAVIS_BUILD_DIR make travis'
```

Comment:
This feels like an awful hack, heh.

Reply:
Yeah it does, I can try alternative ways in separate PRs to fix that later, now that the tests are running to see if I can improve that.